contracts/test_doubles/unit_tests/credit_rule/CustomRuleWithCredit.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity ^0.5.0;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../../../external/SafeMath.sol";
+import "../../../rules/CreditRule.sol";
+
+contract CustomRuleWithCredit {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+    event Pay(
+        address _to,
+        uint256 _amount
+    );
+
+    /* Storage */
+
+    CreditRule public creditRule;
```

Comment:
Should we introduce CreditRuleInterface contract for rules contract to integrate ❓